src/libcore/mem.rs

```diff
@@ -611,3 +611,80 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
+/// representation is. Note that if the enum's representation is signed, then
+/// the discriminant values returned by this function will not compare in the
+/// same order as actual enum variants would under `#[derive(PartialOrd)]`.
+///
+/// If `T` is not an enum, the return value is unspecified (this is **not** undefined
+/// behavior, but the defined behavior is that the value is unpredictable and subject
+/// to change).
+///
+/// # Stability
+///
+/// Discriminants can change if enum variants are reordered, if a new variant is added
+/// in the middle, or (in the case of a C-like enum) if explicitly set discriminants are changed.
```

Comment:
As previously, I would like to not guarantee anything about discriminants in sum-types. i.e. it seems to me that it would be completely fine if discriminants changed even between recompilations of the crate with the same version of the Rust compiler as long as the `discriminant(Enum::Value) == discriminant(Enum::Value)` holds for all dependencies etc.